packages/frontend/src/pages/Executions/index.tsx

```diff
@@ -1,45 +1,84 @@
 import type { IExecution } from '@plumber/types'
 
 import * as React from 'react'
+import { BiSearch } from 'react-icons/bi'
 import { Link, useSearchParams } from 'react-router-dom'
 import { useQuery } from '@apollo/client'
+import {
+  Divider as ChakraDivider,
+  Flex,
+  Icon,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputRightElement,
+} from '@chakra-ui/react'
 import Box from '@mui/material/Box'
 import CircularProgress from '@mui/material/CircularProgress'
 import Divider from '@mui/material/Divider'
-import Grid from '@mui/material/Grid'
 import Pagination from '@mui/material/Pagination'
 import PaginationItem from '@mui/material/PaginationItem'
 import Container from 'components/Container'
 import ExecutionRow from 'components/ExecutionRow'
+import ExecutionStatusMenu from 'components/ExecutionStatusMenu'
 import NoResultFound from 'components/NoResultFound'
 import PageTitle from 'components/PageTitle'
 import { GET_EXECUTIONS } from 'graphql/queries/get-executions'
 import useFormatMessage from 'hooks/useFormatMessage'
+import debounce from 'lodash/debounce'
 
 const EXECUTION_PER_PAGE = 10
 const EXECUTIONS_TITLE = 'Executions'
 
-const getLimitAndOffset = (page: number) => ({
+const getLimitAndOffset = (
+  page: number,
+  filterStatus: string,
+  searchInput: string,
+) => ({
   limit: EXECUTION_PER_PAGE,
   offset: (page - 1) * EXECUTION_PER_PAGE,
+  status: filterStatus,
+  searchInput: searchInput,
 })
 
 export default function Executions(): React.ReactElement {
   const formatMessage = useFormatMessage()
   const [searchParams, setSearchParams] = useSearchParams()
   const page = parseInt(searchParams.get('page') || '', 10) || 1
 
+  const [searchInput, setSearchInput] = React.useState<string>('')
+
+  const handleSearchValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(event.target.value)
+    setSearchParams({ status: filterStatus })
```

Comment:
http://localhost:3001/executions?status=

^ seems a bit weird that no status will give a dangling query param like this.

Reply:
Also, why are you not storing the search string in the query param as well? If i search for a pipe and click into an execution, and pageback, i will end up with different results.

Reply:
initially, i followed the pipes page where they don't store search string in the query params but ok lemme rework this portion to be more ideal